extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java

```diff
@@ -27,18 +27,37 @@
     public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method,
             CallOptions callOptions, Channel next) {
 
-        boolean isOnEventLoop = Context.isOnEventLoopThread();
-        Context context = Vertx.currentContext();
-
         return new ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
 
             @Override
             public void start(Listener<RespT> responseListener, Metadata headers) {
+
+                Context context = Vertx.currentContext();
+                boolean isOnIOThread = context != null && Context.isOnEventLoopThread();
+
                 super.start(new ForwardingClientCallListener.SimpleForwardingClientCallListener<RespT>(responseListener) {
 
+                    @Override
+                    public void onReady() {
```

Comment:
I was under impression that we want to invoke client stuff always with duplicated context but we stick to only propagating the context, right?

Reply:
Yes.
The problem is that if we start using a duplicated context, we cannot block anymore, which would break the blocking clients. So I just propagate if there is a context.